src/jinja2/environment.py

```diff
@@ -433,7 +436,13 @@ def overlay(
 
         return _environment_sanity_check(rv)
 
-    lexer = property(get_lexer, doc="The lexer for this environment.")
+    def lexer_getter(self):
```

Comment:
This change isn't needed. We already have `Environment.lexer`, if you need to inject this during tests override it with a subclass.

```python
class TatsuEnvironment(Environment):
    lexer = TatsuLexer()
```